### PR TITLE
Feature/decisions courts tribunal page with data

### PR DESCRIPTION
**Pull request by timcowlishaw**
## Changes in this PR:

New courts and tribunals page, with all metadata, plus individual court pages  which list the judgments for that court with pagination, and also show a crest and summary blurb. The former is stored in /static/images/court_crests, named with the court param with slashes substituted for underscores, the latter in /static/markdown/court_descriptions, named similarly). Courts without a crest or a description will pull in a default image (Currently the MoJ crest) and a default blurb (currently lorem ipsum) - these should be replaced with something suitable before release. The other option would be to omit crest / description where it is not provided.

I've also removed the dependency on the grid from tna frontend for now as it [wasn't a simple matter to integrate](https://trello.com/c/fD1rbQD2/1741-using-tna-frontend-component-library-for-ui) and gave us very little in this instance that wasn't trivial to implement in CSS. 

## Trello card / Rollbar error (etc)

https://trello.com/c/YlzzrDFX/1707-%F0%9F%8F%A1epic-homepage-changes

## Screenshots of UI changes:
Screenshot 2024-02-27 at 14 01 14
Screenshot 2024-02-27 at 14 01 05
Screenshot 2024-02-27 at 14 00 40

### CODE REVIEW

**Comment by timcowlishaw** [comment]
[🏡EPIC: Homepage changes](https://trello.com/c/YlzzrDFX/1707-%F0%9F%8F%A1epic-homepage-changes)

**Review by Donna-H — APPROVED** [review]

File: config/urls.py [hunk 1]
```diff
@@ -16,6 +16,16 @@
...
     path(settings.ADMIN_URL, admin.site.urls),
     # Your stuff: custom urls includes go here
+    path(
+        "courts-and-tribunals/<path:param>",
+        views.CourtOrTribunalView.as_view(),
+        name="court_or_tribunal",
+    ),
+    path(
+        "courts-and-tribunals",
+        views.CourtsTribunalsView.as_view(),
```
Comment (dragon-dxw) [inline config/urls.py:26]: Would CourtsTribunalsListView be more descriptive?

File: config/views.py [hunk 2]
```diff
@@ -19,6 +25,56 @@ def get_context_data(self, **kwargs):
...
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["context"]["courts"] = courts.get_grouped_selectable_courts()
+        context["context"]["tribunals"] = courts.get_grouped_selectable_tribunals()
+        context["feedback_survey_type"] = "courts_and_tribunals"
+        return context
+
+
+class CourtOrTribunalView(TemplateViewWithContext):
+    template_name = "pages/court_or_tribunal.html"
```
Comment (dragon-dxw) [inline config/views.py:41]: Given the slightly confusingly named classes, doc strings to disambiguate the list from the individual might be nice.

File: ds_judgements_public_ui/static/markdown/court_descriptions/default.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent non viverra dui, quis dignissim metus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. In a risus in tellus suscipit porta. Curabitur risus erat, congue vel lectus non, pellentesque aliquam nulla. Morbi purus lectus, consequat pharetra metus hendrerit, convallis varius neque. Nullam vitae sapien purus. Aenean fermentum quam et nisl pharetra, et gravida dui posuere. In aliquet eros a lacus congue pellentesque eu in risus. Donec non venenatis metus.
```
Comment (dragon-dxw) [inline ds_judgements_public_ui/static/markdown/court_descriptions/default.md:1]: Not loving lorem ipsum text here: an empty document might make more sense.

Comment (timcowlishaw) [reply ds_judgements_public_ui/static/markdown/court_descriptions/default.md:1]: yeah  I think this should go - i had that there for testing but if we don't yet have a generic text, best to just show nothing

File: ds_judgements_public_ui/static/markdown/court_descriptions/uksc.md [hunk 4]
```diff
@@ -0,0 +1 @@
+The UK Supreme Court is the final court for appeals in civil and criminal cases from across the UK. Typically this court deals with important legal questions that affect the public or the constitution. The UK Supreme Court was established in October 2009. You can read more about the role of the court on the [United Kingdom Supreme Court](https://www.supremecourt.uk/){target="\_blank"} website.
```
Comment (dragon-dxw) [inline ds_judgements_public_ui/static/markdown/court_descriptions/uksc.md:1]: I was wondering whether it makes more sense for this to be a part of the utils data about courts; but it's fine here and I don't think we'll want to reuse it elsewhere.

File: ds_judgements_public_ui/static/images/court_crests/default.svg
Comment (dragon-dxw) [inline ds_judgements_public_ui/static/images/court_crests/default.svg:1]: Are we sure that this default crest is acceptable for all courts?

File: ds_judgements_public_ui/templates/pages/courts_and_tribunals.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  {% if context.page_title %}{{ context.page_title }} -{% endif %}
+  Judgements and decisions by court or tribunal
+{% endblock title %}
+{% block content %}
+  <div class="courts-and-tribunals container">
+    <div>
+      <div class="courts-and-tribunals__intro">
+        <h1>Judgments and decisions by court or tribunal</h1>
+        <p>
+          The UK courts and tribunal system is like a legal hierachy.  The Supreme Court is at the top, dealing with the final appeals.
```
Comment (dragon-dxw) [inline ds_judgements_public_ui/templates/pages/courts_and_tribunals.html:16]: Is all text agreed?

File: ds_judgements_public_ui/static/markdown/court_descriptions/uksc.md (hunk 4 above)
Comment (timcowlishaw) [reply ds_judgements_public_ui/static/markdown/court_descriptions/uksc.md:1]: aha yep there was a discussion on slack about this a while back - i think we settled on 'something easily editable in the PUI' just to make the turnaround for edits a bit quicker - this way we don't have to do a utils PR, then a utils version bump PR, then a utils release, then a PUI PR, then a PUI deploy if we want to tweak the text.

File: judgments/templatetags/court_utils.py [hunk 6]
```diff
@@ -53,3 +62,36 @@ def get_court_date_range(court):
...
+@register.filter
+def get_court_judgments_count(court):
+    return search_judgments_and_parse_response(
+        api_client, SearchParameters(court=court.canonical_param)
+    ).total
+
+
+@register.filter
+def get_court_intro_text(court):
+    def read_markdown(param):
```
Comment (dragon-dxw) [inline judgments/templatetags/court_utils.py:76]: I think a happy-case test for markdown feels valuable.

File: ds_judgements_public_ui/static/images/court_crests/default.svg
Comment (timcowlishaw) [reply ds_judgements_public_ui/static/images/court_crests/default.svg:1]: no :-) This is one of the things i was going to check with rose before this goes live!

File: judgments/templatetags/court_utils.py [hunk 7]
```diff
@@ -53,3 +62,36 @@ def get_court_date_range(court):
...
+            with open(path) as file:
+                return md.render(file.read())
+
+    return read_markdown(court.canonical_param) or read_markdown("default")
+
+
+@register.filter
+def get_court_crest_path(court):
+    param = court.canonical_param
+    filename = param.replace("/", "_")
```
Comment (dragon-dxw) [inline judgments/templatetags/court_utils.py:91]: And a happypath test for the crest.

File: judgments/templatetags/court_utils.py (hunk 6 above)
Comment (timcowlishaw) [reply judgments/templatetags/court_utils.py:76]: aha i added some tests on another PR, this one got merged before i could sneak them in. I should've done them before opening the PR really.

File: ds_judgements_public_ui/templates/pages/courts_and_tribunals.html (hunk 5 above)
Comment (timcowlishaw) [reply ds_judgements_public_ui/templates/pages/courts_and_tribunals.html:16]: aha, I *think* this particular text is, but i will check with rose.

File: judgments/templatetags/court_utils.py [hunk 8]
```diff
@@ -53,3 +62,36 @@ def get_court_date_range(court):
...
+    base_path = "images/court_crests/%s.%s"
+    for extension in ["gif", "png", "jpg", "svg"]:
+        path = base_path % (filename, extension)
```
Comment (dragon-dxw) [inline judgments/templatetags/court_utils.py:94]: I'd typically prefer:

```
basepath = r"images/court_crests/{filename}.{extension}"
path = basepath.format(filename=filename, extension=extension)
```

**Pull request merged**